packages/mongodb-runner/src/mongocluster.ts

```diff
@@ -75,17 +232,23 @@ export class MongoCluster extends EventEmitter<MongoClusterEvents> {
       servers: this.servers.map((srv) => srv.serialize()),
       shards: this.shards.map((shard) => shard.serialize()),
       oidcMockProviderProcess: this.oidcMockProviderProcess?.serialize(),
+      defaultConnectionOptions: jsonClone(this.defaultConnectionOptions ?? {}),
+      users: jsonClone(this.users),
     };
   }
 
   isClosed(): boolean {
-    return this.servers.length === 0 && this.shards.length === 0;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const _ of this.children()) return true;
```

Comment:
Should it be returning false if there are children? Not sure I follow the logic here, maybe worth a comment? I might be missing something.

Reply:
This matches the existing logic – if all servers have been removed from the cluster, than it's closed. But I'm happy to add a comment here